easy_my_coop_connector/components/account_payment_adapter.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _
+from odoo.exceptions import ValidationError
+from odoo.fields import Date
+
+from .abstract_emc_adapter import AbstractEMCAdapter
+
+
+class AccountPaymentAdapter(AbstractEMCAdapter):
+    _model = "account.payment"
+    _service = "payment"
+
+    def to_write_values(self, api_dict):
+        api_dict = api_dict.copy()
```

Comment:
No, it should not modify the dictionary: the function `to_write_values` is meant to return a dictionary that can be passed on to `create` . In the case of payment, it is trivial but it gets more complex for other models. Also, I wrote a doc in the abstract class:

```py
    def to_write_values(self, api_dict):
        """
        :return a tuple with
            - the external id
            - a writable dictionary for _model
        received from the api
        """
        raise NotImplementedError
```

I'll fix AccountInvoiceAdapter to avoid unpleasant side effects.

Reply:
And rename dictionnaries.